tests/Sulu/Component/Rest/ListBuilder/Metadata/General/Driver/XmlDriverTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Metadata\General\Driver;
+
+use Metadata\ClassMetadata;
+use Metadata\Driver\FileLocatorInterface;
+use Prophecy\Argument;
+use Sulu\Component\Rest\ListBuilder\Metadata\General\PropertyMetadata;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+
+class XmlDriverTest extends \PHPUnit_Framework_TestCase
+{
+    public function testLoadMetadataFromFileComplete()
+    {
+        $locator = $this->prophesize(FileLocatorInterface::class);
+        $parameterBag = $this->prophesize(ParameterBagInterface::class);
+
+        $driver = new XmlDriver($locator->reveal(), $parameterBag->reveal());
+
+        $reflectionMethod = new \ReflectionMethod(get_class($driver), 'loadMetadataFromFile');
+        $reflectionMethod->setAccessible(true);
+
+        $result = $reflectionMethod->invokeArgs(
+            $driver,
+            [new \ReflectionClass(new \stdClass()), __DIR__ . '/Resources/complete.xml']
+        );
+
+        self::assertInstanceOf(ClassMetadata::class, $result);
+        self::assertEquals('stdClass', $result->name);
+        self::assertCount(5, $result->propertyMetadata);
+
+        self::assertEquals(
+            ['id', 'firstName', 'lastName', 'avatar', 'fullName'],
+            array_keys($result->propertyMetadata)
+        );
+
+        $this->metadataTest($result->propertyMetadata['id'], 'public.id', true, false, 'integer');
+        $this->metadataTest($result->propertyMetadata['firstName'], 'contact.contacts.firstName', false, true);
+        $this->metadataTest($result->propertyMetadata['lastName'], 'contact.contacts.lastName', false, true);
+        $this->metadataTest(
+            $result->propertyMetadata['avatar'],
+            'public.avatar',
+            false,
+            true,
+            'thumbnails',
+            '',
+            '',
+            false
+        );
+        $this->metadataTest(
+            $result->propertyMetadata['fullName'],
+            'public.name',
+            true,
+            false,
+            'string',
+            '100px',
+            '50px',
+            false,
+            false,
+            'test-class'
+        );
```

Comment:
A better way to do this would be to use an array with the expected values:

```php
$expected = array(
    'id' => array(
        'translation' => 'public.id',
        'disabled' => true,
        'default' => false,
        // ...
    ),
    'firstName' => array(
        // ...
    )
);

foreach ($expected as $expectedMetadata) {
    // normalize the data so we can ommit unrequired keys above and set defaults.
    $expectedMetadata = array_merge(array(
        'translation' => null,
        'disabled' => null,
        // ...
    ), $expectedMetadata);
    $this->assertEquals($expectedMetadata['translation'], $metadata->getTranslation();
    $this->assertEquals($expectedMetadata['disabled'], $metadata->getDisabled();
    // ...
}
````

This is easier to understand and maintain.